contracts/core/Allo.sol

```diff
@@ -96,26 +102,30 @@ contract Allo is Initializable, Ownable, MulticallUpgradeable {
     /// @param _registry The address of the registry
     /// @param _treasury The address of the treasury
     /// @param _feePercentage The fee percentage
-    function initialize(address _registry, address payable _treasury, uint256 _feePercentage) public reinitializer(1) {
+    /// @param _baseFee The base fee
+    function initialize(address _registry, address payable _treasury, uint256 _feePercentage, uint256 _baseFee)
+        public
+        reinitializer(1)
+    {
         _initializeOwner(msg.sender);
 
         registry = Registry(_registry);
         treasury = _treasury;
         feePercentage = _feePercentage;
+        baseFee = _baseFee;
 
         emit RegistryUpdated(_registry);
         emit TreasuryUpdated(_treasury);
-        emit FeeUpdated(_feePercentage);
-
-        __Multicall_init();
+        emit FeePercentageUpdated(_feePercentage);
+        emit BaseFeeUpdated(_baseFee);
     }
 
     /// ====================================
     /// =========== Modifier ===============
     /// ====================================
 
-    modifier onlyPoolAdmin(uint256 _poolId) {
-        if (!registry.isOwnerOrMemberOfIdentity(pools[_poolId].identityId, msg.sender)) {
+    modifier onlyPoolOwner(uint256 _poolId) {
+        if (!hasRole(keccak256(abi.encodePacked(_poolId)), msg.sender)) {
```

Comment:
Is there a reason to make this the hash of the pool ID instead of the pool ID itself? It seems nice for it to be legible.

Reply:
> Roles are referred to by their bytes32 identifier. These should be exposed in the external API and be unique. The best way to achieve this is by using public constant hash digests:

Isn't that cause the role has to bytes32 
https://docs.openzeppelin.com/contracts/3.x/api/access#AccessControl